tools/pipelines/templates/include-test-task.yml

```
@@ -15,8 +15,8 @@ parameters:
   default: false
 
 steps:
-# Test - With coverage
-- ${{ if and(parameters.testCoverage, startsWith(parameters.taskTestStep, 'ci:test')) }}:
+# Test - With coverage - Don't run with coverage if the task is a PR and the task is a tinylicious service and stress test.
+- ${{ if and(parameters.testCoverage, startsWith(parameters.taskTestStep, 'ci:test'), or(ne(variables['Build.Reason'], 'PullRequest'), and(not(contains(parameters.taskTestStep, 'realsvc:tinylicious')), not(contains(parameters.taskTestStep, 'stress:tinylicious'))))) }}:
```

It's pretty hard to parse otherwise. I think we also need to say _why_ we're skipping coverage for those tests in PRs (to match what we do in the bundle analysis pipeline, IIRC based on what we discussed, right?).
```suggestion
- ${{ if and(
	parameters.testCoverage,
	startsWith(parameters.taskTestStep, 'ci:test'),
	or(
		ne(variables['Build.Reason'], 'PullRequest'),
		and(
			not(contains(parameters.taskTestStep, 'realsvc:tinylicious')),
			not(contains(parameters.taskTestStep, 'stress:tinylicious'))
		)
	)
) }}:
```

---

Added the comment, but indentation wise, it needs to be in 1 line otherwise yaml complain about the structure and does not recognize it as a condition.

---

Hmm did it complain both with tabs and spaces? We have some [working examples](https://github.com/microsoft/FluidFramework/blob/fcd15e0cb38d037b6f3dfbd4a36f7850190a16d1/tools/pipelines/build-docs.yml#L73-L76) of expressions spread across more than 1 line, but I couldn't find one for an `if` expression. If it doesn't work, oh well.